### PR TITLE
Updated Makefile with deploy target

### LINKED ISSUE #4: Enable e2e tests for Azure
We should set up an e2e tests for this repo

- [x] Setup e2e tests infra on Azure
- [x] Enable e2e tests for PRs
- [x] Switch to test-infra for presubmit e2e tests (presubmit tests are running for pull requests in cloud-provider-azure repo) @ritazh 
  - [x] Add e2e steps to setup test-infra via Makefile
  - [x] Update presubmit tests to the new way
  - [x] Remove current perl scripts
- [ ] Enable full periodic e2e tests for stable releases (periodic tests are running in backgroud for each release braches)
  - [ ] Add more test scenarios, e.g. conformance, correctness, alpha-features, autoscaling, scalability, multi-zone, slow and serial
  - [ ] Enable e2e for more releases, e.g. v1.13 and v1.12
  - [ ] Setup prow jobs for new test scenarios
- [ ] Add Azure features e2e testings #7
  - [ ] LoadBalancer service tests with various annotations
  - [ ] ACR image pulling tests without docker secrets setting explicitly
  - [ ] Multi availability zones tests
  - [ ] Cloud provider configuration options tests (e.g. Standard LoadBalancer and VMSS)
  - [ ] Upgrading tests (e.g. cloud-provider version updates)
- [ ] Documentation for e2e @ritazh

### PR DESCRIPTION
**Pull request by Abhik1998**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This PR adds a new target to the makefile named deploy.
**Which issue(s) this PR fixes**:

Fixes #4 

**Special notes for your reviewer**:

**Release note**:
```
action-required: review

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,9 +25,13 @@ TEST_RESULTS_DIR=testResults
 GOMETALINTER_OPTION=--tests --disable-all -E gofmt -E vet -E golint
 
 IMAGE_REGISTRY ?= local
-K8S_VERSION ?= v1.14.0
+K8S_VERSION ?= v1.14.0-rc.1
```
Comment (andyzhangx) [inline Makefile]: why it's still `v1.14.0-rc.1`?

File: hack/deploy-cluster.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-# Copyright 2019 The Kubernetes Authors.
+# Copyright 2018 The Kubernetes Authors.
```
Comment (andyzhangx) [inline hack/deploy-cluster.sh]: change back to 2019

File: Makefile (hunk 1 above)
Comment (feiskyer) [reply Makefile]: Please rollback this line, v1.14.0 should be used here

File: hack/deploy-cluster.sh [hunk 3]
```diff
@@ -56,12 +56,12 @@ SUB6=$(echo $SUB5| cut -d"}" -f 1)
 TENANT_ID=${TENANT_ID:-"$SUB6"}
 
 
-echo "Type the filename of the generated api-model to be used in .json format"
-read FILENAME
+#echo "Type the filename of the generated api-model to be used in .json format"
+FILENAME=$1
```
Comment (feiskyer) [inline hack/deploy-cluster.sh]: Please check whether FILENAME is set here

File: Makefile [hunk 4]
```diff
@@ -25,9 +25,13 @@ TEST_RESULTS_DIR=testResults
...
 AKSENGINE_VERSION ?= master
 HYPERKUBE_IMAGE ?= "gcrio.azureedge.net/google_containers/hyperkube-amd64:$(K8S_VERSION)"
+
+CCM_IMAGE_NAME=custom-cloud-controller-manager
+CCM_IMAGE_TAG ?= $(shell git rev-parse --short=7 HEAD)
+CCM_IMAGE=$(IMAGE_REGISTRY)/$(CCM_IMAGE_NAME):$(CCM_IMAGE_TAG)
 # manifest name under tests/e2e/k8s-azure/manifest
```
Comment (feiskyer) [inline Makefile]: These 3 lines are not required, we should use `IMAGE` instead

File: Makefile [hunk 5]
```diff
@@ -117,3 +121,12 @@ test-e2e: image hyperkube
 
 test-ccm-e2e:
 	go test ./tests/e2e/ -timeout 0 -v $(CCM_E2E_ARGS)
+
+deploy: image hyperkube
```
Comment (feiskyer) [inline Makefile]: Add another `docker push $(IMAGE)` before deploy

File: Makefile [hunk 6]
```diff
@@ -117,3 +121,12 @@ test-e2e: image hyperkube
...
+        docker build $(CCM_IMAGE)\
```
Comment (feiskyer) [inline Makefile]: image target would build the image, so this line should be removed

File: Makefile [hunk 7]
```diff
@@ -117,3 +121,12 @@ test-e2e: image hyperkube
...
+            --build-arg K8S_VERSION=$(K8S_VERSION) \
+		    --build-arg AKSENGINE_VERSION=$(AKSENGINE_VERSION) \
+		    tests/k8s-azure
+		docker push $(CCM_IMAGE)
+		touch	TEMPFILE	|	sed	's/$(IMAGE)/$(CCM_IMAGE)/'	/tests/k8s-azure/manifests/$(TEST_MANIFEST)	>TEMPFILE	\
```
Comment (feiskyer) [inline Makefile]: we can move the file changes to hack/deploy-cluster.sh and let the script accept $(IMAGE)

File: Makefile (hunk 1 above)
Comment (Abhik1998) [reply Makefile]: Okay Sir I am updating that

File: hack/deploy-cluster.sh (hunk 2 above)
Comment (Abhik1998) [reply hack/deploy-cluster.sh]: Ok sorry Sir

File: Makefile [hunk 8]
```diff
@@ -28,6 +28,7 @@ IMAGE_REGISTRY ?= local
 K8S_VERSION ?= v1.14.0
 AKSENGINE_VERSION ?= master
 HYPERKUBE_IMAGE ?= "gcrio.azureedge.net/google_containers/hyperkube-amd64:$(K8S_VERSION)"
+TEMPFILE=$(mktemp)
```
Comment (feiskyer) [inline Makefile]: let's move this into hack/deploy-cluster.sh

File: Makefile [hunk 9]
```diff
@@ -117,3 +121,8 @@ test-e2e: image hyperkube
 
 test-ccm-e2e:
 	go test ./tests/e2e/ -timeout 0 -v $(CCM_E2E_ARGS)
+
+deploy: push image hyperkube\
+		hack/deploy-cluster.sh	tests/k8s-azure/manifests/$(TEST_MANIFEST) $(TEMPFILE)
```
Comment (feiskyer) [inline Makefile]: should be:

```
deploy: image hyperkube push
    IMAGE=$(IMAGE) HYPERKUBE_IMAGE=$(HYPERKUBE_IMAGE) hack/deploy-cluster.sh

```

File: hack/deploy-cluster.sh [hunk 10]
```diff
@@ -56,12 +56,30 @@ SUB6=$(echo $SUB5| cut -d"}" -f 1)
...
+TEMPFILE=$2
+if [ "$FILENAME" = "" ]
```
Comment (feiskyer) [inline hack/deploy-cluster.sh]: please use `if [ -z "$var" ]` instead

File: Makefile [hunk 11]
```diff
@@ -117,3 +119,8 @@ test-e2e: image hyperkube
 
 test-ccm-e2e:
 	go test ./tests/e2e/ -timeout 0 -v $(CCM_E2E_ARGS)
+
+deploy: image hyperkube	push
+	IMAGE=$(IMAGE) HYPERKUBE_IMAGE=$(HYPERKUBE_IMAGE) hack/deploy-cluster.sh
+	hack/deploy-cluster.sh	tests/k8s-azure/manifests/$(TEST_MANIFEST)
```
Comment (feiskyer) [inline Makefile]: please remove lines 125, 126

**Pull request merged**